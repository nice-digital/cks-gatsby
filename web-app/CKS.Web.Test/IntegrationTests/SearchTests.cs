using CKS.Web.Test.IntegrationTests.Infrastructure;
using Moq;
using Newtonsoft.Json;
using NICE.Search.Common.Interfaces;
using NICE.Search.Common.Models;
using Shouldly;
using System.IO;
using System.Net;
using System.Net.Http;
using Xunit;
using Xunit.Abstractions;

namespace CKS.Web.Test.IntegrationTests
{
	public class SearchTests : TestBase
	{
		public SearchTests(ITestOutputHelper output) : base(output)
		{ }


		[Fact]
		public async void ReturnsJson()
		{
			// Arrange
			var client = _factory
				.WithFakeSearchProvider("FakeSearchResult-Cancer.json")
				.CreateClient();

			// Act
			var response = await client.GetAsync("/api/search");
			var responseBody = await response.Content.ReadAsStringAsync();

			// Assert
			response.StatusCode.ShouldBe(System.Net.HttpStatusCode.OK);
			response.Content.Headers.ContentType.MediaType.ShouldBe("application/json");
			responseBody.ShouldMatchApproved();
		}
	}
}
