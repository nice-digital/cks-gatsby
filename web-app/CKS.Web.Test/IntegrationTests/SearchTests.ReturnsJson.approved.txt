{
  "debug": {
    "rawResponse": "Valid NEST response built from a successful low level call on POST: /cks/document/_search?typed_keys=true\r\n# Audit trail of this API call:\r\n - [1] HealthyResponse: Node: http://localhost:9200/ Took: 00:00:01.0265052\r\n# Request:\r\n{\"from\":0,\"size\":10,\"highlight\":{\"pre_tags\":[\"<mark>\"],\"post_tags\":[\"</mark>\"],\"fields\":{\"content\":{\"fragment_size\":150,\"no_match_size\":150,\"type\":\"fvh\",\"matched_fields\":[\"content\",\"content.plain\"]},\"title\":{\"pre_tags\":[\"<b>\"],\"post_tags\":[\"</b>\"],\"fragment_size\":300,\"number_of_fragments\":0,\"type\":\"fvh\",\"matched_fields\":[\"title\",\"title.plain\"]}}},\"_source\":{\"excludes\":[\"content\"]},\"aggs\":{\"unique_count\":{\"cardinality\":{\"field\":\"clusterid\"}}},\"query\":{\"function_score\":{\"query\":{\"function_score\":{\"query\":{\"bool\":{\"should\":[{\"bool\":{\"should\":[{\"bool\":{\"must\":[{\"query_string\":{\"type\":\"best_fields\",\"query\":\"CANCER\",\"default_operator\":\"and\",\"analyzer\":\"fulltext_query\",\"analyze_wildcard\":true,\"fields\":[\"content^1\",\"title^1\",\"metadescription^0\",\"keywords^1\"],\"tie_breaker\":1.0,\"quote_field_suffix\":\".plain\",\"auto_generate_synonyms_phrase_query\":false}}]}},{\"bool\":{\"must\":[{\"query_string\":{\"type\":\"best_fields\",\"query\":\"CANCER\",\"default_operator\":\"and\",\"analyzer\":\"fulltext_no_synonyms\",\"analyze_wildcard\":true,\"fields\":[\"content^1\",\"title^1\",\"metadescription^0\",\"keywords^1\"],\"tie_breaker\":1.0,\"quote_field_suffix\":\".plain\",\"auto_generate_synonyms_phrase_query\":false}}]}},{\"bool\":{\"must\":[{\"multi_match\":{\"type\":\"phrase\",\"query\":\"CANCER\",\"analyzer\":\"standard\",\"fields\":[\"title.plain\",\"content.plain\",\"metadescription.plain\"]}}]}}]}}]}},\"functions\":[{\"field_value_factor\":{\"field\":\"boostvalue\",\"factor\":0.01}},{\"gauss\":{\"freshness_date\":{\"scale\":\"28d\",\"offset\":\"28d\"}}}],\"boost_mode\":\"multiply\"}},\"functions\":[{\"weight\":1.0}],\"score_mode\":\"multiply\"}}}\r\n# Response:\r\n{\"took\":286,\"timed_out\":false,\"_shards\":{\"total\":1,\"successful\":1,\"skipped\":0,\"failed\":0},\"hits\":{\"total\":3,\"max_score\":16.007385,\"hits\":[{\"_index\":\"cks_integration_tests\",\"_type\":\"document\",\"_id\":\"FYZ39HEBujuv9LwEFlM6\",\"_score\":16.007385,\"_source\":{\"_test\":\"BooleanOrQueriesShouldWork\",\"nicedoctype\":[\"Guidance\"],\"boostvalue\":100,\"clusterid\":-100016,\"title\":\"cancer\",\"url\":\"https://www.nice.org.uk\",\"source_name\":\"test\",\"landing_url\":\"https://www.nice.org.uk\"},\"highlight\":{\"title\":[\"<b>cancer</b>\"],\"content\":[\"\"]}},{\"_index\":\"cks_integration_tests\",\"_type\":\"document\",\"_id\":\"FoZ39HEBujuv9LwEFlM6\",\"_score\":11.551083,\"_source\":{\"_test\":\"BooleanOrQueriesShouldWork\",\"boostvalue\":100,\"clusterid\":-100017,\"title\":\"\",\"url\":\"https://www.nice.org.uk\",\"source_name\":\"test\",\"landing_url\":\"https://www.nice.org.uk\"},\"highlight\":{\"content\":[\"<mark>cancer</mark>\"]}},{\"_index\":\"cks_integration_tests\",\"_type\":\"document\",\"_id\":\"EoZ39HEBujuv9LwEFlM6\",\"_score\":10.060528,\"_source\":{\"_test\":\"StopWordsInSynonynms\",\"boostvalue\":100,\"clusterid\":-100013,\"title\":\"benign enlargement of prostate\",\"url\":\"https://www.nice.org.uk\",\"source_name\":\"test\",\"landing_url\":\"https://www.nice.org.uk\"},\"highlight\":{\"content\":[\"prostatic <mark>cancer</mark>\"]}}]},\"aggregations\":{\"cardinality#unique_count\":{\"value\":3}}}\r\n"
  },
  "failed": false,
  "errorMessage": null,
  "parentResultUrl": null,
  "resultCount": 3,
  "rolledUpCount": 3,
  "unrolledCount": 3,
  "documents": [
    {
      "id": "FYZ39HEBujuv9LwEFlM6",
      "tagsHigh": null,
      "guidanceRef": null,
      "xml": null,
      "abstract": null,
      "metaDescription": null,
      "accredited": false,
      "alternativePublicationDate": null,
      "altSourceNames": [],
      "areaOfInterest": null,
      "breadcrumbs": null,
      "childDocumentCount": 0,
      "childDocuments": [],
      "childDocumentsUrl": null,
      "clusterId": -100016,
      "evidenceTypes": [],
      "impact": null,
      "content": null,
      "contentId": "https://www.nice.org.uk",
      "index": 0,
      "lessSuitableForPrescribing": false,
      "niceDocType": [
        "Guidance"
      ],
      "niceGuidanceType": [],
      "niceAdviceType": [],
      "niceResultType": null,
      "parentPathTitle": null,
      "parentPathUrl": null,
      "parentPathWayUrl": null,
      "pathAndQuery": "/",
      "publicationDate": null,
      "lastUpdated": null,
      "resourceType": [],
      "guidanceStatus": [],
      "publicationType": null,
      "publisher": null,
      "resourceCategory": null,
      "relatedLinks": null,
      "relatedLinkList": {},
      "sourceName": "test",
      "sourceUrl": "https://www.nice.org.uk",
      "staticAbstract": null,
      "subSectionLinks": null,
      "summaryUrl": null,
      "tags": [],
      "teaser": "",
      "title": "<b>cancer</b>",
      "titleNoHtml": null,
      "boostValue": 100,
      "score": 16.007385,
      "titleParts": [
        "<b>cancer</b>"
      ],
      "ukmiSummaryUrl": null,
      "url": "https://www.nice.org.uk",
      "debug": null
    },
    {
      "id": "FoZ39HEBujuv9LwEFlM6",
      "tagsHigh": null,
      "guidanceRef": null,
      "xml": null,
      "abstract": null,
      "metaDescription": null,
      "accredited": false,
      "alternativePublicationDate": null,
      "altSourceNames": [],
      "areaOfInterest": null,
      "breadcrumbs": null,
      "childDocumentCount": 0,
      "childDocuments": [],
      "childDocumentsUrl": null,
      "clusterId": -100017,
      "evidenceTypes": [],
      "impact": null,
      "content": null,
      "contentId": "https://www.nice.org.uk",
      "index": 0,
      "lessSuitableForPrescribing": false,
      "niceDocType": [],
      "niceGuidanceType": [],
      "niceAdviceType": [],
      "niceResultType": null,
      "parentPathTitle": null,
      "parentPathUrl": null,
      "parentPathWayUrl": null,
      "pathAndQuery": "/",
      "publicationDate": null,
      "lastUpdated": null,
      "resourceType": [],
      "guidanceStatus": [],
      "publicationType": null,
      "publisher": null,
      "resourceCategory": null,
      "relatedLinks": null,
      "relatedLinkList": {},
      "sourceName": "test",
      "sourceUrl": "https://www.nice.org.uk",
      "staticAbstract": null,
      "subSectionLinks": null,
      "summaryUrl": null,
      "tags": [],
      "teaser": "<mark>cancer</mark>",
      "title": "",
      "titleNoHtml": null,
      "boostValue": 100,
      "score": 11.551083,
      "titleParts": [
        ""
      ],
      "ukmiSummaryUrl": null,
      "url": "https://www.nice.org.uk",
      "debug": null
    },
    {
      "id": "EoZ39HEBujuv9LwEFlM6",
      "tagsHigh": null,
      "guidanceRef": null,
      "xml": null,
      "abstract": null,
      "metaDescription": null,
      "accredited": false,
      "alternativePublicationDate": null,
      "altSourceNames": [],
      "areaOfInterest": null,
      "breadcrumbs": null,
      "childDocumentCount": 0,
      "childDocuments": [],
      "childDocumentsUrl": null,
      "clusterId": -100013,
      "evidenceTypes": [],
      "impact": null,
      "content": null,
      "contentId": "https://www.nice.org.uk",
      "index": 0,
      "lessSuitableForPrescribing": false,
      "niceDocType": [],
      "niceGuidanceType": [],
      "niceAdviceType": [],
      "niceResultType": null,
      "parentPathTitle": null,
      "parentPathUrl": null,
      "parentPathWayUrl": null,
      "pathAndQuery": "/",
      "publicationDate": null,
      "lastUpdated": null,
      "resourceType": [],
      "guidanceStatus": [],
      "publicationType": null,
      "publisher": null,
      "resourceCategory": null,
      "relatedLinks": null,
      "relatedLinkList": {},
      "sourceName": "test",
      "sourceUrl": "https://www.nice.org.uk",
      "staticAbstract": null,
      "subSectionLinks": null,
      "summaryUrl": null,
      "tags": [],
      "teaser": "prostatic <mark>cancer</mark>",
      "title": "benign enlargement of prostate",
      "titleNoHtml": null,
      "boostValue": 100,
      "score": 10.060528,
      "titleParts": [
        "benign enlargement of prostate"
      ],
      "ukmiSummaryUrl": null,
      "url": "https://www.nice.org.uk",
      "debug": null
    }
  ],
  "pagerLinks": {
    "previous": null,
    "first": null,
    "pages": [
      {
        "title": "1",
        "url": {
          "route": null,
          "af": null,
          "am": null,
          "arms": null,
          "ci": null,
          "cid": null,
          "co": null,
          "em": null,
          "fc": null,
          "fd": null,
          "from": null,
          "fs": null,
          "fullUrl": "?",
          "m": null,
          "om": null,
          "pa": null,
          "pathTerm": null,
          "pbu": null,
          "ps": null,
          "pt": null,
          "pwbu": null,
          "q": null,
          "rf": null,
          "s": null,
          "sa": null,
          "aggsOnly": null,
          "sm": null,
          "sn": null,
          "sp": null,
          "spuid": null,
          "sr": null,
          "ss": null,
          "syn": null,
          "t": null,
          "td": null,
          "title": null,
          "to": null,
          "report": null,
          "debug": null
        },
        "isCurrent": true
      }
    ],
    "next": null
  },
  "pageSize": 10,
  "firstResult": 1,
  "lastResult": 3,
  "finalSearchText": "cancer",
  "finalSearchTextNoStopWords": "cancer",
  "originalSearch": null,
  "navigators": [],
  "unfilteredResultsUrl": null
}